include/ppx/ui_util.h

```diff
@@ -0,0 +1,102 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ppx_moving_average_h
+#define ppx_moving_average_h
+
+#include <cmath>
+#include <cstdint>
+#include <type_traits>
+
+#include "ppx/log.h"
+#include "ppx/timer.h"
+
+namespace ppx {
+
+// A weighted moving average where w_i = e^{(t_i-t_{now})/halfLife}
+
+template <typename FloatT>
+class ExponentialDecayingMovingAverage
+{
+public:
+    ExponentialDecayingMovingAverage() = default;
+
+    void ClearHistory() { mAccWeight = 0; }
```

Comment:
should `mMean` and `mAccVar` be reset to 0 as well?

Reply:
Ah, I probably need to fix the mAccVar calculation.
`mMean` is not valid immediately after clear history since there's no sample, and will be changed to new value due to `mAccWeight` being 0.

Reply:
Switched to a different algorithm, since it does not need to deal with sqrt of negative caused by floating point and it's more versatile in case we want to use if for other statistics calculation.